### PR TITLE
PRMP-749 - Bulk Upload: Reject deceased patients

### PR DESCRIPTION
**Pull request by joefong-nhs**

### CODE REVIEW

File: lambdas/services/mock_data/pds_patient_9000000001_X4S4L_pcse.json [hunk 1]
```diff
@@ -73,7 +73,6 @@
     "gender": "female",
     "birthDate": "2010-10-22",
     "multipleBirthInteger": 1,
-    "deceasedDateTime": "2010-10-22T00:00:00+00:00",
```
Comment (joefong-nhs) [inline lambdas/services/mock_data/pds_patient_9000000001_X4S4L_pcse.json:76]: Here I removed the deceased status from all our test patients, except 201 and 202 which are added to test ingesting deceased patients.

File: lambdas/models/pds_models.py [hunk 2]
```diff
@@ -183,4 +228,13 @@ def get_minimum_patient_details(self, nhs_number) -> PatientDetails:
...
+            deceased=is_deceased(death_notification_status),
+            deathNotificationStatus=death_notification_status,
         )
+
+
+def is_deceased(death_notification_status: Optional[DeathNotificationStatus]) -> bool:
+    return (
+        death_notification_status == DeathNotificationStatus.FORMAL
+        or death_notification_status == DeathNotificationStatus.INFORMAL
+    )
```
Comment (steph-torres-nhs) [inline lambdas/models/pds_models.py:240]: is it possible to use date/time deceased instead?

File: lambdas/services/bulk_upload_service.py [hunk 3]
```diff
@@ -111,9 +112,23 @@ def handle_sqs_message(self, message: dict):
...
+                )
+                raise PatientDeceasedException(
+                    f"Patient is deceased - {death_notification_status}"
+                )
+
+        except (
+            LGInvalidFilesException,
+            PatientRecordAlreadyExistException,
+            PatientDeceasedException,
+        ) as error:
```
Comment (steph-torres-nhs) [inline lambdas/services/bulk_upload_service.py:131]: potential to extract into a new function to help with readability?

File: lambdas/services/bulk_upload_service.py [hunk 4]
```diff
@@ -111,9 +112,23 @@ def handle_sqs_message(self, message: dict):
...
                 raise LGInvalidFilesException("Patient not registered at your practice")
-        except (LGInvalidFilesException, PatientRecordAlreadyExistException) as error:
+
+            if pds_patient_details.deceased:
+                death_notification_status = (
+                    pds_patient_details.death_notification_status
+                )
+                raise PatientDeceasedException(
+                    f"Patient is deceased - {death_notification_status}"
+                )
```
Comment (steph-torres-nhs) [inline lambdas/services/bulk_upload_service.py:125]:
```suggestion
            if pds_patient_details.deceased:
               
                raise PatientDeceasedException(
                    f"Patient is deceased - {pds_patient_details.death_notification_status.name}"
                )
```

**Review by steph-torres-nhs — APPROVED** [review]

**Pull request merged**